rest/route/project.go

```diff
@@ -748,3 +750,76 @@ func (h *getProjectVersionsHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return resp
 }
+
+type testProjectAliasHandler struct {
+	sc data.Connector
+
+	version             string
+	alias               string
+	includeDependencies bool
+	project             *dbModel.Project
+}
+
+func makeTestProjectAliasHandler(sc data.Connector) gimlet.RouteHandler {
+	return &testProjectAliasHandler{
+		sc: sc,
+	}
+}
+
+func (h *testProjectAliasHandler) Factory() gimlet.RouteHandler {
+	return &testProjectAliasHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *testProjectAliasHandler) Parse(ctx context.Context, r *http.Request) error {
+	params := r.URL.Query()
+
+	h.version = params.Get("version")
+	if h.version == "" {
+		return errors.New("'version' parameter must be specified")
+	}
+	h.alias = params.Get("alias")
+	if h.alias == "" {
+		return errors.New("'alias' parameter must be specified")
+	}
+	h.includeDependencies = (params.Get("include_deps") != "")
+
+	v, err := dbModel.VersionFindOneId(h.version)
+	if err != nil {
+		return errors.Errorf("version %s not found", h.version)
+	}
+	pp, err := dbModel.ParserProjectFindOneById(h.version)
```

Comment:
This should also use h.sc, but also this isn't strictly the right thing to do since old versions don't have parser projects (although maybe it's been long enough and we don't care?). This should be LoadProjectForVersion (which also handles 799 through 812; the purpose of ParserProject was to avoid having to marshal/unmarshal.)

Also I think this chunk belongs in Run, not Parse, since it's part of the work of the route and doesn't involve parsing user input.

Reply:
Moved to run and switched to a different function